statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -2894,6 +2894,85 @@ def zvalues(self):
         """
         return self.params / self.bse
 
+    def get_rsquared(self, baseline="rwdrift", **kwargs):
+        # calc sse finite estimation
+        d = np.maximum(self.loglikelihood_burn, self.nobs_diffuse)
+        srss = np.sum(self.standardized_forecasts_error[:, d:]**2, axis=1).T
+        f_T = np.array([self.forecasts_error_cov[i, i, -1]
+                       for i in range(self.model.k_endog)])
+        sse = f_T * srss
+        endog = self.model.endog
+
+        # calc rsquared wrt different baseline models
+        if baseline == "rwdrift":
+            ssdm = np.var(np.diff(endog, axis=0), axis=0) * (len(endog) - 1)
+            rsquared = 1. - sse / ssdm
+        elif baseline == "mean":
+            ssm = np.var(endog, axis=0) * len(endog)
+            rsquared = 1. - sse / ssm
+        elif baseline == "seasonal":
+            from statsmodels.regression.linear_model import OLS
+            from statsmodels.tools.tools import add_constant
+            if 'seasonal' not in kwargs:
+                raise ValueError(
+                    "kwarg seasonal required for rsquared_seasonal.")
+            seasonal = kwargs['seasonal']
+            x = np.zeros((len(endog), seasonal-1))
+            idx = [(i, j) for j in range(seasonal-1)
+                   for i in range(j, len(endog), seasonal)]
+            x[tuple(np.array(idx).T)] = 1
+            ssdsm = np.zeros(self.model.k_endog)
+            for i in range(self.model.k_endog):
+                seasonalmodel = OLS(endog[:, i], add_constant(x)).fit()
+                ssdsm[i] = seasonalmodel.ssr
+            rsquared = 1. - sse / ssdsm
+        else:
+            raise NotImplementedError(
+                "In state space models there is not a single baseline " +
+                "model that will always be useful for rsquared computation. " +
+                "baseline must be in ['rwdrift', 'mean', 'seasonal']" +
+                "It is recommended to use MLEResult.rsquared_rwdrift "
+                )
+        # return float if k_endog == 1
+        if self.model.k_endog == 1:
+            return rsquared[0]
+        else:
+            return rsquared
+
+    def rsquared(self):
+        """
+        In state space models there is not a single comparison
+        model that will always be useful for rsquared computation.
+        However, we have implmented `rsquared_rwdrift` and `rsquared_mean`
+        It is recommended to use `rsquared_rwdrift`
+        """
+        return self.get_rsquared('')
```

Comment:
Doesn't this produce an error?

Reply:
If the error is intentional, you should directly raise NotImplementedError here. You can pull the common message outside the class to share it.